src/triage/component/postmodeling/contrast/model_evaluator.py

```diff
@@ -0,0 +1,1155 @@
+        merged_df = pd.merge(mat,
+                             self.predictions,
+                             on=['entity_id', 'as_of_date'],
+                             how='inner',
+                             suffixes=('test', 'pred'))
+
+        cache[path] = merged_df
+        return merged_df
+
+    def train_matrix(self, path):
+        '''
+        Load predictions matrices using the catwalk.storage.ProjectStorage.
+        This class allow the user to define a project path that can be either a
+        system local path or an s3 path and it will handle itself this
+        different infrastructures.
+
+        Once defined, we can pass this object to the MatrixStorageEngine that
+        will read this object and return a MatrixSotre object with a set of
+        handy methods to handle matrices
+
+        Arguments:
+            path: project path to initiate the ProjectStorage object
+        '''
+        cache = self.__dict__.setdefault('_preds_matrix_cache', {})
+        try:
+            return cache[path]
+        except KeyError:
+            pass
+
+        storage = ProjectStorage(path)
+        matrix_storage = MatrixStorageEngine(storage).get_store(self.train_matrix_uuid)
+        mat = matrix_storage.matrix
+
+        # Merge with predictions table and return complete matrix
+        merged_df = mat.merge(self.predictions,
+                             on='entity_id',
+                             how='inner',
+                             suffixes=('test', 'pred'))
+
+        cache[path] = mat
+        return mat
+
+    def plot_score_distribution(self,
+                               save_file=False,
+                               name_file=None,
+                               figsize=(16,12),
+                               fontsize=20):
+        '''
+        Generate an histograms with the raw distribution of the predicted
+        scores for all entities.
+            - Arguments:
+                - save_file (bool): save file to disk as png. Default is False.
+                - name_file (string): specify name file for saved plot.
+                - label_names(tuple): define custom label names for class.
+                - figsize (tuple): specify size of plot.
+                - fontsize (int): define custom fontsize. 20 is set by default.
+
+        '''
+
+        df_ = self.predictions.filter(items=['score'])
+
+        fig, ax = plt.subplots(1, figsize=figsize)
+        plt.hist(df_.score,
+                 bins=20,
+                 normed=True,
+                 alpha=0.5,
+                 color='blue')
+        plt.axvline(df_.score.mean(),
+                    color='black',
+                    linestyle='dashed')
+        ax.set_ylabel('Frequency', fontsize=fontsize)
+        ax.set_xlabel('Score', fontsize=fontsize)
+        plt.title('Score Distribution', y =1.2,
+                  fontsize=fontsize)
+        plt.show()
+
+        if save_file:
+            plt.savefig(str(name_file + '.png'))
```

Comment:
Rather than allow the caller to raise a `TypeError` with the invocation --

    plot_score_distribution(save_file=True)

-- perhaps either:

* remove the `save_file` Boolean and only save when a truthy `name_file` is supplied
* don't bother with saving (or perhaps even with showing) on their behalf -- they can easily:

        plot = model_evaluator.plot_score_distribution()
        plot.show()
        plot.savefig(...)